### ISSUE TITLE
High system CPU usage when idle

### ISSUE THREAD

**Issue by maxocoin** [greymass/anchor#988]
I have around 15 accounts set up on Anchor and even when  anchor is just idle, it eats around  of 25% of the CPU on my system (Intel Core i5-6300 at 2.4 GHz and 16 GB of RAM)

![image](https://user-images.githubusercontent.com/30597258/93331604-74c20f80-f818-11ea-874a-2a521fe030fd.png)

I assume it is the continuous update in the background for any changes in the account

Couldn't that be stopped after some time without interaction with Anchor?

**Comment by aaroncox** [comment]
Yeah, it's continuously updating accounts, balances, etc. We have considered throttling it when it's in the background, but we're worried it would fall too far behind and lag a lot worse when it comes back into focus.

As of the 1.1.x series, you can now actually just close the main Anchor window when you're not using it. Anchor now runs and lives in the system tray (down by the clock) and can respond to signing requests, and if you click the logo, it'll open the main window back up. That currently is the best option when it comes to reducing the load Anchor puts on a system when it's not in use. 

We'll continue to push to try and optimize it further though, it's a bit silly right now.

**Comment by maxocoin** [comment]
Great!!, looks like "closing" the main window does the trick.

But I think that, unless they are needed for sing a transaction, all those "refreshes" should be placed on a low priority thread and not slow down the UI

The hourglass indications are perfect and you could even add the ability to click on it, and show a modal "updating..." window for give priority to a specific account

**Comment by aaroncox** [comment]
With 15 accounts at a 15 second interval (which it's currently set to), it currently takes almost 4 minutes to cycle through all of the loaded accounts you have. The massive amount of API calls comes from the fact that for each of the 15 accounts, we need to make an individual API call for each token you're tracking. So if you have an additional 20 tokens tracked, that's 20 additional API calls the wallet needs to make to also check those balances, for each account (so 300 for all 15 accounts).

The APIs are the core of the problem here, which is something we're setting our sights on improving as we move forward. We'll see if we can maybe set some sort of throttling option at the very least, because Anchor becomes pretty rough when you have a lot of accounts and tokens being tracked.

**Comment by maxocoin** [comment]
Is all that information needed for signing?  If not, those calls could be all done in  a  low priority thread, for not slowing down the signing process (the most important part of Anchor IMO)

Another option could be implement #934 and allow an easy switch between smaller set of wallets.

O add an option to increase this 15 seconds interval or increase it automatically based on the number of accounts  configured on Anchor?  In my case, I don't care if the info in not up to date, as long as the "hourglass" makes me aware of it.

**Comment by Saltant** [comment]
You can use this microservice for temporary fix zombie processes: https://github.com/Saltant/anchor-patch
It runs in the background and kills zombie processes if they appear.
